Issues/#8454 document theme builder am (#861)
gravitee-io/gravitee-docs: pages/am/3.x/user-guide/branding/user-guide-branding-internationalization.adoc

= Internationalization
:page-sidebar: am_3_x_sidebar
:page-permalink: am/current/am_userguide_branding_internationalization.html
:page-folder: am/user-guide
:page-layout: am

[label label-version]#New in version 3.19#

== Overview

> AM is aiming to support multiple languages for the end user convenience.

Reviewer: ```suggestion
AM supports internationalization in multiple languages so that end users can benefit from a great user experience.
```

> AM is aiming to support multiple languages for the end user convenience.

Reviewer: AM supports internationalization in multiple languages so that end users can benefit from a great user experience.

> The internationalization option is available under domain `*Design -> Texts*` section

Reviewer: ```suggestion
The internationalization option is available under the domain `*Design -> Texts*` section.
```

Author: Force pushed

> NOTE: Under the hood all the templates uses Thymeleaf and Freemarker engines to support translation.

Reviewer: ```suggestion
NOTE: Under the hood, all templates use Thymeleaf and Freemarker engines to support translation.
```

image::{% link images/am/current/graviteeio-am-userguide-branding-theme-builder-texts.png %}[Texts example,650,500]

== Customize translation

> NOTE: *3.19.0* release supports English and French language out of the box.

Reviewer: ```suggestion
NOTE: The *3.19.0* release supports English and French languages out of the box.
```

> To create a new language or customise texts for a default supported languages:

Reviewer: ```suggestion
To create a new language or customize texts for a default supported language:
```

. link:{{ '/am/current/am_userguide_authentication.html' | relative_url }}[Log in to AM Console^].
. Click *Settings > Texts*.

> . Click `*ADD A NEW LANGUAGE*` button in the `*Languages*` section

Reviewer: ```suggestion
. Click `*ADD A NEW LANGUAGE*` button in the `*Languages*` section.
```

image::{% link images/am/current/graviteeio-am-userguide-branding-theme-builder-add-language.png %}[Theme builder,250,100]

[start=4]

> . Select a language code from the drop-down menue and click `*ADD*` button.

Reviewer: ```suggestion
. Select a language code from the dropdown menu and click the `*ADD*` button.
```

> . Click `*SAVE CHANGES*` button.

Reviewer: ```suggestion
. Click the `*SAVE CHANGES*` button.
```

> . Select the language from the drop-down in the `*Translations*` section

Reviewer: ```suggestion
. Select the language from the dropdown menu in the `*Translations*` section.
```

> . Click `*ADD A NEW TRANSLATION*`

Reviewer: ```suggestion
. Click `*ADD A NEW TRANSLATION*`.
```

image::{% link images/am/current/graviteeio-am-userguide-branding-theme-builder-new-translation.png %}[Theme builder,250,100]

[start=8]

> . Provide a valid property key name and value you want to customise

Reviewer: ```suggestion
. Provide a valid property key name and value that you want to customize.
```

> . Click `*ADD*` and then `*SAVE CHANGES*`

Reviewer: ```suggestion
. Click `*ADD*` and then `*SAVE CHANGES*`.
```

> NOTE: The change should be immediately viewable in the `*Theme*` preview section for the template.

Reviewer: ```suggestion
NOTE: You should be able to view the change immediately in the `*Theme*` preview section for the template.
```

== Limitations

> * Default locale is based on user web browser, the requested language cannot be overridden with query param such as *ui_locales*.

Reviewer: ```suggestion
* The default locale is based on the end user's web browser - the requested language cannot be overridden with a query parameter such as *ui_locales*.
```

> == Out of the box translated properties

Reviewer: ```suggestion
== Out-of-the-box translated properties
```

> Default forms and email templates contain a set of translated properties which are listed below.

Reviewer: ```suggestion
The default forms and email templates contain a set of translated properties, which are listed below.
```

> NOTE: For example, if you want to support Spanish language and translate the Login page title,

Reviewer: ```suggestion
NOTE: For example, if you want to support Spanish and translate the Login page title,
```
